packages/aws-rfdk/lib/lambdas/nodejs/sep-configuration/handler.ts

```diff
@@ -0,0 +1,288 @@
+import { SecretsManager } from 'aws-sdk';
+import { LambdaContext } from '../lib/aws-lambda';
+import { CfnRequestEvent, SimpleCustomResource } from '../lib/custom-resource';
+import { DeadlineClient } from '../lib/deadline-client';
+import {
+  isArn as isSecretArn,
+  Secret,
+} from '../lib/secrets-manager';
+import { EventPluginRequests } from '../lib/sep-configuration';
+
+export interface IConnectionOptions {
+  /**
+   * FQDN of the host to connect to.
+   */
+  readonly hostname: string;
+
+  /**
+   * Port on the host.
+   */
+  readonly port: string;
+
+  /**
+   * Protocol used to connect to the host.
+   */
+  readonly protocol: string;
+
+  /**
+   * Content of the CA certificate
+   */
+  readonly caCertificate?: string;
+
+  /**
+   * Content of the PFX certificate.
+   */
+  readonly pfxCertificate?: string;
+
+  /**
+   * Shared passphrase used for a single private key and/or a PFX.
+   */
+  readonly passphrase?: string;
+}
+
+/**
+ * The input to this Custom Resource
+ */
+export interface ISEPConfiguratorResourceProperties {
+  /**
+   * Connection info for logging into the server.
+   */
+  readonly connection: IConnectionOptions;
+
+  /**
+   * A JSON string containing the Spot Fleet Request Configurations.
+   * See https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/event-spot.html?highlight=spot%20even%20plugin#example-spot-fleet-request-configurations
+   */
+  readonly spotFleetRequestConfigurations?: object;
+
+  /**
+   * The Spot Event Plugin settings.
+   * See https://docs.thinkboxsoftware.com/products/deadline/10.1/1_User%20Manual/manual/event-spot.html?highlight=spot%20even%20plugin#event-plugin-configuration-options
+   */
+  readonly spotPluginConfigurations?: object;
+}
+
+/**
+ * A custom resource used to save Spot Event Plugin server data and configurations.
+ */
+export class SEPConfiguratorResource extends SimpleCustomResource {
+  protected readonly secretsManagerClient: SecretsManager;
+
+  constructor(secretsManagerClient: SecretsManager) {
+    super();
+    this.secretsManagerClient = secretsManagerClient;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public validateInput(data: object): boolean {
+    return this.implementsISEPConfiguratorResourceProperties(data);
+  }
+
+  /**
+   * @inheritdoc
+   */
+  // @ts-ignore -- we do not use the physicalId
+  public async doCreate(physicalId: string, resourceProperties: ISEPConfiguratorResourceProperties): Promise<object|undefined> {
+    const eventPluginRequests = await this.spotEventPluginRequests(resourceProperties.connection);
+
+    if (resourceProperties.spotFleetRequestConfigurations) {
+      const stringConfigs = this.spotFleetRequestToString(resourceProperties.spotFleetRequestConfigurations);
+      const response = await eventPluginRequests.saveServerData(stringConfigs);
+      if (!response) {
+        console.log(`Failed to save spot fleet request with configuration: ${stringConfigs}`);
+      }
```

Comment:
First I though that maybe we don't want to fail the whole deployment because of configurations, that can be easily added later in the spot event plugin manually. But now I think it makes more sense to throw an error here.

Reply:
Yeah, this makes more sense. As a customer, I'd expect that my changes deploy correctly or that an error is surfaced. That is now addressed - thanks! Resolved.